python/tvm/topi/riscv_cpu/conv2d_int8.py

```diff
@@ -0,0 +1,362 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name,unused-variable,unused-argument,no-member
+"""Conv2D int8 schedule on RISCV"""
+from tvm import te, target, autotvm
+from ..utils import traverse_inline, get_const_tuple
+from ..generic import conv2d as conv2d_generic
+from .. import nn
+from ..nn.conv2d import _get_workload as _get_conv2d_workload, unpack_NCHWc_to_nchw
+from ..x86.conv2d_int8 import _pack_data
+from ..nn.utils import get_pad_tuple
+from .tensor_intrin import dot_int8_int8_int32, int8_conv2d_impl
+
+
+def _get_default_config(cfg, data, kernel, strides, padding, dilation, out_dtype):
+    """
+    Get default int8 schedule config for the workload
+    """
+    wkl = _get_conv2d_workload(data, kernel, strides, padding, dilation, out_dtype)
+    is_kernel_1x1 = wkl.kernel_h == 1 and wkl.kernel_w == 1
+    if is_kernel_1x1:
+        conv2d_generic.fallback_schedule_cpu_1x1_int8(cfg, wkl, int32_lanes=4, num_int8_elements=4)
+    else:
+        conv2d_generic.fallback_schedule_cpu_common_int8(
+            cfg, wkl, int32_lanes=4, num_int8_elements=4
+        )
+
+
+def is_int8_hw_support(data_dtype, kernel_dtype):
+    """
+    Checks to ensure that we can use int8 on riscv_cpu.
+    1) The datatypes are correct.
+    2) The vector extension "V" is used.
+    """
+    # 1) Check datatypes.
+    is_dtype_support = data_dtype == "uint8" and kernel_dtype == "int8"
+
+    # 2) Check target.
+    current_target = target.Target.current(allow_none=False)
+    has_attr = "+v" in current_target.mattr
+    is_arch_support = "v" in current_target.arch[2:]
+    if not is_arch_support and "march" in current_target.attrs:
+        is_arch_support = "v" in current_target.attrs["march"]
```

Comment:
I was not aware that `-march=` is actually exposed to the TVM target. This is great because it would also work with non-llvm TVM targets.